dbcrossbar/src/clouds/gcloud/client.rs

```diff
@@ -234,20 +315,12 @@ impl Client {
         Output: fmt::Debug + DeserializeOwned,
         U: IntoUrl + fmt::Debug,
         Query: fmt::Debug + Serialize,
-        Body: fmt::Debug + Serialize,
+        Body: fmt::Debug + Serialize + Sync + Send,
     {
         let url = build_url(url, query)?;
         trace!("POST {} {:?}", url, body);
         trace!("serialied {}", serde_json::to_string(&body)?);
-        let token = self.token().await?;
-        let http_resp = self
-            .client
-            .post(url.as_str())
-            .bearer_auth(token.as_str())
-            .json(&body)
-            .send()
-            .await
-            .with_context(|| format!("could not POST {}", url))?;
+        let http_resp = self.post_with_retry_helper(&url, &body).await?;
```

Comment:
We definitely don't want to plug into the main POST logic here—this would make us retry _every_ POST to Google Cloud, not just `CREATE VIEW`. And some of those posts won't be idempotent, so this may break stuff elsewhere in `dbcrossbar`'s BigQuery or GCS drivers.

We want a BigQuery-level Rust API that says "Run this SQL statement with a possible retry," basically.

Reply:
I'm really tempted for `post` to take a parameter:

```
enum Idempotency {
    SafeToRetry,
    UnsafeToRetry,
}
```

We might then be able to check this directly inside the `wait`.

Probably we should inline the helper but factor out response->WaitStatus conversion (shared with GET).